### PR TITLE
[doc] Activate the nitpicky option in sphinx build

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :scroll: Docs          |

## Description

This will warn about all missing references

### CODE REVIEW

**Comment by Pierre-Sassoulas** [comment]
Labeling high effort because it look like we need cross referencing with the astroid doc.

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas This is ready for review 😄

File: doc/development_guide/contribute.rst [hunk 1]
```diff
@@ -71,9 +71,9 @@ Tips for Getting Started with Pylint Development
   codebase and will help you identify where you will need to make changes
   for what you are trying to implement.
 
-* :func:`astroid.extract_node` is your friend. Most checkers are AST based,
-  so you will likely need to interact with :mod:`astroid`.
-  A short example of how to use :func:`astroid.extract_node` is given
+* ``astroid.extract_node`` is your friend. Most checkers are AST based,
+  so you will likely need to interact with ``astroid``.
+  A short example of how to use ``astroid.extract_node``` is given
```
Comment (Pierre-Sassoulas) [inline doc/development_guide/contribute.rst:76]:
```suggestion
  A short example of how to use ``astroid.extract_node`` is given
```

**Comment by Pierre-Sassoulas** [comment]
You're right, cross referencing between two doc would be a nice to have but we don't need that right now.

**Pull request merged**